### DISCUSSION TITLE
Better similar position depending going up or down

### DISCUSSION THREAD

**Discussion by Pulpyyyy** [nagyrobi/home-assistant-custom-components-cover-rf-time-based#30] [Ideas]
Hello,

Currently, if i do request a 40% position, the reached position of the cover is not physicaly identical if starting from 0% or starting from 100%.
No matter time values setted for ​​up or down.

Is it possible to add a fixed time value which will be added when goes from 0 -> 1% or 1 -> 0% (that way, cover is reaching 'clear path' at 1% every time ) ?

- 0% -> 100% = fix time + up time
- 0% -> X% = fix time + (up time * X%)
- 0% -> 1% = fix time
- 1% -> X% = up time * X%
- X% -> 1% = down time * X%
- 1% -> 0% = fix time
- 100% -> 0% = fix time + (down time  * X%)
- 
Regards

**Comment by nagyrobi** [comment]
Asked many times, not easy to implement.

**Comment by Pulpyyyy** [comment]
Could some kind of template value be used for time_up... in config file (assuming it is dynamicaly loaded enough) : 
`      travelling_time_up: >-
        {% set travel=15 %}
        {% set fix=7 %}
        {% if state_attr("cover.test","current_position") == 0 -%}
          {{ travel + fix}}
        {%- else -%}
          {{ travel}}
        {%- endif %}`
Any new position is at least 1% ....

But it's much more complicated for downtime... Requested position is needed at least...
